### PR TITLE
Merge runtime / runtime-only

### PR DESCRIPTION
**Pull request by pgherveou**

### CODE REVIEW

File: integration-tests/e2e-runtime-only-backend/lib.rs [hunk 1]
```diff
@@ -156,7 +156,7 @@ pub mod flipper {
         }
 
         /// Just instantiate a contract using non-default runtime.
-        #[ink_e2e::test(backend = "runtime_only", runtime = ink_e2e::MinimalRuntime)]
+        #[ink_e2e::test(backend(runtime_only = ink_e2e::MinimalRuntime))]
```
Comment (SkymanOne) [inline integration-tests/e2e-runtime-only-backend/lib.rs:159]: So, I would rather have 
```suggestion
        #[ink_e2e::test(backend(drink, runtime = ink_e2e::MinimalRuntime))]
```

File: integration-tests/e2e-runtime-only-backend/lib.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ pub mod flipper {
         /// - flip the flipper
         /// - get the flipper's value
         /// - assert that the value is `true`
-        #[ink_e2e::test(backend = "runtime_only")]
+        #[ink_e2e::test(backend(runtime_only = ::drink::MinimalRuntime))]
```
[the remark this answers is not in the document]
Comment (pgherveou) [reply integration-tests/e2e-runtime-only-backend/lib.rs:87]: I think you need to write some custom darling attributes for that to work.
https://github.com/TedDriggs/darling/issues/249#issuecomment-1655595079

File: integration-tests/e2e-runtime-only-backend/lib.rs (hunk 1 above)
Comment (pgherveou) [reply integration-tests/e2e-runtime-only-backend/lib.rs:159]: The easier here to avoid writing darling custom parsing code would be to use a `Drink{Runtime: Option<Syn::Path>}`
so that you can write `backend(drink())` for default and `backend(drink(foo))` for a `Some(foo)` value

File: integration-tests/e2e-runtime-only-backend/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (pgherveou) [reply integration-tests/e2e-runtime-only-backend/lib.rs:87]: actually see comment below for maybe a better compromise

**Comment by pgherveou** [comment]
@SkymanOne updated. 
tldr the associated type force you to at least specify the backend like this to use the default runtime

```rust
 #[ink_e2e::test(backend(runtime_only()))]
```

specifying a custom one is done with
```rust
 #[ink_e2e::test(backend(runtime_only(runtime = ink_e2e::MinimalRuntime)))]
```


e2e tests run by default or I need to trigger them manually, to check nothing is broken?

**Review by pmikolajczyk41 — APPROVED** [review]
very good idea, thanks!
